generated/nidcpower/session.py

```diff
@@ -8,6 +8,41 @@
 from nidcpower import library_singleton
 from nidcpower import visatype
 
+import pprint
+pp = pprint.PrettyPrinter(indent=4)
+
+
+class _TimedeltaConverter(object):
+    def __init__(self, value, library_type, library_units):
+        self._value = value
+        self._library_type = library_type
+        self._library_units = library_units
+
+        if str(type(value)).find("'datetime.timedelta'") != -1:
+            if library_units == 'seconds':
+                scaling = 1
+            elif library_units == 'milliseconds':
+                scaling = 1000
+            elif library_units == 'microseconds':
+                scaling = 1000000
+            else:
+                raise TypeError("units must be 'seconds', 'milliseconds', or 'microseconds'. Actual {0}".format(library_units))
+
+            self.value = value.total_seconds() * scaling
+        else:
```

Comment:
This allows customer to pass either the original type or the timedelta... I don't think it's worth the flexibility at this point. I'd rather accept just the type.

Reply:
I disagree. It seems like it is overkill to force them to create a timedelta if they don't plan on using it.